enterprise/releases/2023-06-0.mdx

```diff
@@ -0,0 +1,77 @@
+- Issue e-mail notifications: Include pganalyze organization name in email body, for easier filtering when using multiple organizations
+- Add additional telemetry data to check pganalyze snapshot health via Prometheus/Grafana
+  - See [instructions](/docs/enterprise/settings#worker-queue-monitoring-using-prometheus) for how to use the Prometheus endpoint
+- Allow setting `SIDEKIQ_CONCURRENCY=0` to turn off workers in combined Enterprise image
+  - You can utilize this setting in order to separate the workers from the container servicing the pganalyze web UI, avoiding slow web requests caused by worker processing
+  - Note such a setup requires having configured object storage as well as an external Redis server (set via `REDIS_URL`)
+  - See [all settings](/docs/enterprise/settings)
+
+
+## Performance
+
+- Query analysis: Speed up analysis by avoiding to process schema dependency tracking
+- Improve performance of getIssues by avoiding N+1 queries
+- Disable use of prepared statements for web application
+  - In practice this ends up causing more overhead than benefit for the application, and we've seen better results without prepared statements
+- Cache table names associated to each query directly on `queries` table
+  - This speeds up background tasks working on a per-table basis (e.g. Index Advisor), as well as per-table queries list in the web application
+- Background workers:
+  - Don't run check-up logic for servers that have not received data in the last 24 hours
+  - Speed up truncation of queries (truncated queries are used for display in tables)
+  - Batch check-up logic together for 10 database servers at a time
+
+
+## Bugfixes
+
+- Object storage: Fix DNS lookup errors from Rust-based snapshot workers
+  - This may have shown as an I/O error that caused no snapshot data to be processed, despite the collector successfully sending data to the installation
+  - If we have previously asked you to utilize an IP address for your MinIO endpoint as a workaround, you can now switch back to using a DNS hostname
+- Query analysis:
+  - Don't retry query analysis after 'relation does not exist' errors
+  - Avoid "Unexpected rtekind" errors for functions and VALUES
+  - Fix planning for hash indexes (this allows Index Advisor to run on tables that have hash indexes, but does not yet recommend hash indexes)
+- Remove query length limit for parsing
+  - Previously query texts exceeding 20,000 characters were not parsed during the initial intake, causing them to not have referenced tables, or an associated statement type. This is change removes this limit.
+- getIssues API: Fix legacy displayName field to actually return the name, not the UUID
+  - Note that the displayName field is deprecated and replaced by `references`.`name` (see the current [getIssues API](/docs/api/queries/getIssues) docs)
+- Explain text to JSON: Add support for "Presorted Key" property
```

Comment:
Can you explain which PR is associated with this? I think I'm not so aware of this change and I don't really know which one you're talking about (though it might just be lack of product knowledge of me)

Reply:
I think it's https://github.com/pganalyze/pganalyze/pull/2783 -- but "Explain text to JSON" may not mean much to our users. Maybe instead something like

> Fix parsing of explain plans using the "Presorted Key" property when using `auto_explain.log_format = text`"

?

Reply:
Ohh, thanks for the pointer. Yeah I think Maciek's proposal is more easy to understand, so +1 in general.

`auto_explain.log_format` is `text` by default so maybe it's okay to drop "when using..." part? Or, "when using text format auto_explain"?
Also technically it's a fix indeed (as it's introduced 13 and it's been a while), I think "add support" is not so wrong either. So something like:

> Add support for parsing of explain plans using the "Presorted Key" property
> Fix parsing of explain plans using the "Presorted Key" property when using text format auto_explain

(just throwing more suggestions, I'm fine with Maciek's suggestion too)

Reply:
Went with Maciek's suggestion - thanks!

Reply:
> auto_explain.log_format is text by default so maybe it's okay to drop "when using..." part?

That's the default format, but we do recommend setting it to JSON since the text format is pretty ad-hoc and our transformation is not perfect. I'm -0.1 on "when using text format auto_explain": it flows better, but I don't think that's critical in a changelog, and I do think it's nice to mention the concrete setting we're referring to. And I do think "fix" makes more sense than "add support" here, since it was not something we knew didn't work.

(I know this is all moot now since this was just merged; just trying to hone our collective changelog game :smile: ).